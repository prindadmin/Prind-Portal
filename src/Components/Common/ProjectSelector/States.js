export const QUIESCENT = 'QUIESCENT'
export const LOADING = 'LOADING'
export const LOADING_ERROR = 'LOADING_ERROR'
export const UPDATE_ERROR = 'UPDATE_ERROR'
export const PROJECT_TYPE_SELECTOR_OPEN = 'PROJECT_TYPE_SELECTOR_OPEN'
