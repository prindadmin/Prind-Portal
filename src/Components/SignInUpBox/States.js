export const QUIESCENT = 'QUIESCENT'
export const LOADING = 'LOADING'
export const PASSWORD_RESET_SUCCESS = 'PASSWORD_RESET_SUCCESS'
export const PASSWORD_RESET_FAILED = 'PASSWORD_RESET_FAILED'
