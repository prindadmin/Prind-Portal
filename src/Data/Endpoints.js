export const DEFAULTPAGE = ""
export const DEFAULTLOGGEDINPAGE = "/inception"
export const PROJECTDETAILSPAGE = "/project"
export const SIGNINPAGE = "/signin"
export const SIGNUPPAGE = "/signup"
export const MEMBERPAGE = "/team"
