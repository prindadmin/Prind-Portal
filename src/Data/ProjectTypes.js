import * as Strings from './Strings'

const ProjectTypesList = [
  {
    id: 'CDM2015Project',
    icon: 'building',
    name: Strings.PROJECT_TYPE_CDM2015PROJECT_NAME,
    description: Strings.PROJECT_TYPE_CDM2015PROJECT_DESCRIPTION
  },
  {
    id: 'DHSFProject',
    icon: 'building',
    name: Strings.PROJECT_TYPE_DHSFPROJECT_NAME,
    description: Strings.PROJECT_TYPE_DHSFPROJECT_DESCRIPTION
  }
]

export default ProjectTypesList
