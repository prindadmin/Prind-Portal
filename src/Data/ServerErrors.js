
export const PROJECT_NOT_FOUND = "PROJECT_NOT_FOUND"
export const INSUFFICIENT_PERMISSION = "INSUFFICIENT_PERMISSION"
