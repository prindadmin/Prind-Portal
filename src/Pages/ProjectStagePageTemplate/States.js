export const QUIESCENT = 'QUIESCENT'

export const CONTENT_FETCH_IN_PROGRESS = 'CONTENT_FETCH_IN_PROGRESS'
export const CONTENT_FETCH_SUCCESSFUL = 'CONTENT_FETCH_SUCCESSFUL'
export const CONTENT_FETCH_FAILED = 'CONTENT_FETCH_FAILED'
