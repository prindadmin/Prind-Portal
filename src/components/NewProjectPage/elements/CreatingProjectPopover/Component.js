import React, { Component } from 'react'
//import PropTypes from 'prop-types'

import PopOverHandler from '../../../common/popOverHandler'

import {
  Spinner,
  Intent,
} from '@blueprintjs/core'

import * as strings from '../../../../data/Strings'

export class Element extends Component {
  static propTypes = {
  }

  getContent = () => {
    return (
      <React.Fragment>
        {strings.PROJECT_CREATION_BILLING_WARNING}
        <Spinner size='150' intent={Intent.DANGER}/>
      </React.Fragment>
    )
  }

  render () {
    return (
      <PopOverHandler>
        <div id='popup-greyer' onClick={(e) => e.stopPropagation()}>
          <div id='project-creation-popover'>
            <div id='popup-box'>
              <div className='project-creation-popover-container' onClick={(e) => e.stopPropagation()}>
                <div className='element-title'>
                  {strings.PROJECT_CREATION_IN_PROGRESS}
                </div>
                <div className='element-description'>
                  {this.getContent()}
                </div>
              </div>
            </div>
          </div>
        </div>
      </PopOverHandler>
    )
  }

}

export default Element
