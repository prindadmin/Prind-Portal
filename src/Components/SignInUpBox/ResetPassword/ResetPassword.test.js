
import { shallow, mount } from 'enzyme';
import Component from './ResetPassword';
import * as States from '../../../States'
import * as Endpoints from '../../../Data/Endpoints'
import ReactGA from 'react-ga';

import CanUseWebP from '../../../Functions/CheckIfWebpSupported'
jest.mock('../../../Functions/CheckIfWebpSupported')


it('Should render', () => {
  ReactGA.initialize('dummy', { testMode: true });
  const mockToggleVisibleForm = jest.fn()
  const mockSetNewPassword = jest.fn()
  const mockHistoryPush = jest.fn()
  const props = {
    toggleVisibleForm: mockToggleVisibleForm,
    setNewPassword: mockSetNewPassword,
    history: {
      push: mockHistoryPush,
    },
    location: {
      search: "client_id=fbss4knsc8gmgct526ci8kp3a&user_name=892107a6-b223-4774-8dff-0b6fe5414ffe&confirmation_code=587828",
      pathname: "reset-password?",
    }
  }
  const component = shallow(<Component {...props} />);
  expect(component).toMatchSnapshot();
});


it('Should render with webp', () => {
  CanUseWebP.mockReturnValue(true)
  ReactGA.initialize('dummy', { testMode: true });
  const mockToggleVisibleForm = jest.fn()
  const mockSetNewPassword = jest.fn()
  const mockHistoryPush = jest.fn()
  const props = {
    toggleVisibleForm: mockToggleVisibleForm,
    setNewPassword: mockSetNewPassword,
    history: {
      push: mockHistoryPush,
    },
    location: {
      search: "client_id=fbss4knsc8gmgct526ci8kp3a&user_name=892107a6-b223-4774-8dff-0b6fe5414ffe&confirmation_code=587828",
      pathname: "reset-password?",
    }
  }
  const component = shallow(<Component {...props} />);
  expect(component).toMatchSnapshot();
});


it('Submit password change', () => {
  ReactGA.initialize('dummy', { testMode: true });
  const mockToggleVisibleForm = jest.fn()
  const mockSetNewPassword = jest.fn()
  const mockHistoryPush = jest.fn()
  const props = {
    toggleVisibleForm: mockToggleVisibleForm,
    setNewPassword: mockSetNewPassword,
    history: {
      push: mockHistoryPush,
    },
    location: {
      search: "client_id=fbss4knsc8gmgct526ci8kp3a&user_name=892107a6-b223-4774-8dff-0b6fe5414ffe&confirmation_code=587828",
      pathname: "reset-password",
    }
  }
  const component = shallow(<Component {...props} />);
  component.find('#password').simulate('change', {target: {name: 'password', value: 'Password123!'}})
  component.find('#confirmPassword').simulate('change', {target: {name: 'confirmPassword', value: 'Password123!'}})
  component.find('#submit').simulate('click', { preventDefault: () => {}} )
  expect(mockSetNewPassword).toHaveBeenCalledWith(
    {
      password: "Password123!",
      client_id: "fbss4knsc8gmgct526ci8kp3a",
      user_name: "892107a6-b223-4774-8dff-0b6fe5414ffe",
      confirmation_code: "587828"
    },
    expect.any(Function),
    expect.any(Function)
  );
  expect(component).toMatchSnapshot();
});


it('Submit password change but passwords don\'t match', () => {
  ReactGA.initialize('dummy', { testMode: true });
  const mockToggleVisibleForm = jest.fn()
  const mockSetNewPassword = jest.fn()
  const mockHistoryPush = jest.fn()
  const props = {
    toggleVisibleForm: mockToggleVisibleForm,
    setNewPassword: mockSetNewPassword,
    history: {
      push: mockHistoryPush,
    },
    location: {
      search: "client_id=fbss4knsc8gmgct526ci8kp3a&user_name=892107a6-b223-4774-8dff-0b6fe5414ffe&confirmation_code=587828",
      pathname: "reset-password",
    }
  }
  const component = shallow(<Component {...props} />);
  component.find('#password').simulate('change', {target: {name: 'password', value: 'These do'}})
  component.find('#confirmPassword').simulate('change', {target: {name: 'confirmPassword', value: 'not match'}})
  component.find('#submit').simulate('click', { preventDefault: () => {}} )
  expect(component).toMatchSnapshot();
});


it('Submit password change and respond with success', () => {
  ReactGA.initialize('dummy', { testMode: true });
  const mockToggleVisibleForm = jest.fn()
  const mockSetNewPassword = jest.fn((values, resolve, reject) => { resolve() })
  const mockHistoryPush = jest.fn()
  const props = {
    toggleVisibleForm: mockToggleVisibleForm,
    setNewPassword: mockSetNewPassword,
    history: {
      push: mockHistoryPush,
    },
    location: {
      search: "client_id=fbss4knsc8gmgct526ci8kp3a&user_name=892107a6-b223-4774-8dff-0b6fe5414ffe&confirmation_code=587828",
      pathname: "reset-password",
    }
  }
  const component = shallow(<Component {...props} />);
  component.find('#password').simulate('change', {target: {name: 'password', value: 'Password123!'}})
  component.find('#confirmPassword').simulate('change', {target: {name: 'confirmPassword', value: 'Password123!'}})
  component.find('#submit').simulate('click', { preventDefault: () => {}} )
  expect(mockSetNewPassword).toHaveBeenCalledWith(
    {
      password: "Password123!",
      client_id: "fbss4knsc8gmgct526ci8kp3a",
      user_name: "892107a6-b223-4774-8dff-0b6fe5414ffe",
      confirmation_code: "587828"
    },
    expect.any(Function),
    expect.any(Function)
  );
  expect(component).toMatchSnapshot();
});


it('Submit password change and respond with failure', () => {
  ReactGA.initialize('dummy', { testMode: true });
  const mockToggleVisibleForm = jest.fn()
  const mockSetNewPassword = jest.fn((values, resolve, reject) => { reject({ result: { message: "failed"} }) } )
  const mockHistoryPush = jest.fn()
  const props = {
    toggleVisibleForm: mockToggleVisibleForm,
    setNewPassword: mockSetNewPassword,
    history: {
      push: mockHistoryPush,
    },
    location: {
      search: "client_id=fbss4knsc8gmgct526ci8kp3a&user_name=892107a6-b223-4774-8dff-0b6fe5414ffe&confirmation_code=587828",
      pathname: "reset-password",
    }
  }
  const component = shallow(<Component {...props} />);
  component.find('#password').simulate('change', {target: {name: 'password', value: 'Password123!'}})
  component.find('#confirmPassword').simulate('change', {target: {name: 'confirmPassword', value: 'Password123!'}})
  component.find('#submit').simulate('click', { preventDefault: () => {}} )
  expect(mockSetNewPassword).toHaveBeenCalledWith(
    {
      password: "Password123!",
      client_id: "fbss4knsc8gmgct526ci8kp3a",
      user_name: "892107a6-b223-4774-8dff-0b6fe5414ffe",
      confirmation_code: "587828"
    },
    expect.any(Function),
    expect.any(Function)
  );
  expect(component).toMatchSnapshot();
});


it('Click form to get test coverage', () => {
  ReactGA.initialize('dummy', { testMode: true });
  const mockToggleVisibleForm = jest.fn()
  const mockSetNewPassword = jest.fn((values, resolve, reject) => { reject({ result: { message: "failed"} }) } )
  const mockHistoryPush = jest.fn()
  const mockPreventDefault = jest.fn()
  const props = {
    toggleVisibleForm: mockToggleVisibleForm,
    setNewPassword: mockSetNewPassword,
    history: {
      push: mockHistoryPush,
    },
    location: {
      search: "client_id=fbss4knsc8gmgct526ci8kp3a&user_name=892107a6-b223-4774-8dff-0b6fe5414ffe&confirmation_code=587828",
      pathname: "reset-password",
    }
  }
  const component = shallow(<Component {...props} />);
  component.find('#reset-password-form').simulate('submit', { preventDefault: mockPreventDefault } )
  expect(mockPreventDefault).toHaveBeenCalled()
});


it('Click sign in form text', () => {
  ReactGA.initialize('dummy', { testMode: true });
  const mockToggleVisibleForm = jest.fn()
  const mockSetNewPassword = jest.fn((values, resolve, reject) => { resolve() })
  const mockHistoryPush = jest.fn()
  const mockPreventDefault = jest.fn()
  const props = {
    toggleVisibleForm: mockToggleVisibleForm,
    setNewPassword: mockSetNewPassword,
    history: {
      push: mockHistoryPush,
    },
    location: {
      search: "client_id=fbss4knsc8gmgct526ci8kp3a&user_name=892107a6-b223-4774-8dff-0b6fe5414ffe&confirmation_code=587828",
      pathname: "reset-password",
    }
  }
  const component = shallow(<Component {...props} />);
  component.find('#password').simulate('change', {target: {name: 'password', value: 'Password123!'}})
  component.find('#confirmPassword').simulate('change', {target: {name: 'confirmPassword', value: 'Password123!'}})
  component.find('#submit').simulate('click', { preventDefault: () => {}} )
  expect(mockSetNewPassword).toHaveBeenCalledWith(
    {
      password: "Password123!",
      client_id: "fbss4knsc8gmgct526ci8kp3a",
      user_name: "892107a6-b223-4774-8dff-0b6fe5414ffe",
      confirmation_code: "587828"
    },
    expect.any(Function),
    expect.any(Function)
  );
  component.find('#submit').simulate('click', { preventDefault: mockPreventDefault } )
  expect(mockHistoryPush).toHaveBeenCalledWith(Endpoints.SIGNINPAGE)
});
