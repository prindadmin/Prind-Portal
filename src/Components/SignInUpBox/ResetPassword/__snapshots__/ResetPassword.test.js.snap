// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should render 1`] = `
<Fragment>
  <div
    className="logo-container"
  >
    <a
      href="https://buildingim.com"
      rel="noopener noreferrer"
      target="_blank"
    >
      <img
        alt="BuildingIM logo"
        src="/images/logos/buildingim-logo.png"
      />
    </a>
  </div>
  <div
    className="welcome-text-heading"
  >
    Welcome
  </div>
  <div
    className="welcome-text-body"
  >
    Please enter a new password for your account.
  </div>
  <form
    className="sign-in-form form"
    id="reset-password-form"
    onSubmit={[Function]}
  >
    <input
      className="default"
      id="password"
      name="password"
      onChange={[Function]}
      placeholder="Password"
      type="password"
      value=""
    />
    <input
      className="default"
      id="confirmPassword"
      name="confirmPassword"
      onChange={[Function]}
      placeholder="Repeat Password"
      type="password"
      value=""
    />
    <div
      className="spacer"
    />
    <input
      className="submit-button"
      id="submit"
      onClick={[Function]}
      type="submit"
      value="Change Password"
    />
  </form>
</Fragment>
`;

exports[`Should render with webp 1`] = `
<Fragment>
  <div
    className="logo-container"
  >
    <a
      href="https://buildingim.com"
      rel="noopener noreferrer"
      target="_blank"
    >
      <img
        alt="BuildingIM logo"
        src="/images/logos/buildingim-logo.webp"
      />
    </a>
  </div>
  <div
    className="welcome-text-heading"
  >
    Welcome
  </div>
  <div
    className="welcome-text-body"
  >
    Please enter a new password for your account.
  </div>
  <form
    className="sign-in-form form"
    id="reset-password-form"
    onSubmit={[Function]}
  >
    <input
      className="default"
      id="password"
      name="password"
      onChange={[Function]}
      placeholder="Password"
      type="password"
      value=""
    />
    <input
      className="default"
      id="confirmPassword"
      name="confirmPassword"
      onChange={[Function]}
      placeholder="Repeat Password"
      type="password"
      value=""
    />
    <div
      className="spacer"
    />
    <input
      className="submit-button"
      id="submit"
      onClick={[Function]}
      type="submit"
      value="Change Password"
    />
  </form>
</Fragment>
`;

exports[`Submit password change 1`] = `
<Fragment>
  <div
    className="logo-container"
  >
    <a
      href="https://buildingim.com"
      rel="noopener noreferrer"
      target="_blank"
    >
      <img
        alt="BuildingIM logo"
        src="/images/logos/buildingim-logo.png"
      />
    </a>
  </div>
  <div
    className="welcome-text-heading"
  >
    Welcome
  </div>
  <div
    className="welcome-text-body"
  >
    Please enter a new password for your account.
  </div>
  <div
    className="lds-ring"
  >
    <div />
    <div />
    <div />
    <div />
  </div>
</Fragment>
`;

exports[`Submit password change and respond with failure 1`] = `
<Fragment>
  <div
    className="logo-container"
  >
    <a
      href="https://buildingim.com"
      rel="noopener noreferrer"
      target="_blank"
    >
      <img
        alt="BuildingIM logo"
        src="/images/logos/buildingim-logo.png"
      />
    </a>
  </div>
  <div
    className="welcome-text-heading"
  >
    Welcome
  </div>
  <div
    className="welcome-text-body"
  >
    Please enter a new password for your account.
  </div>
  <div
    className="error-callout"
  >
    <div />
  </div>
  <form
    className="sign-in-form form"
    id="reset-password-form"
    onSubmit={[Function]}
  >
    <input
      className="filled"
      id="password"
      name="password"
      onChange={[Function]}
      placeholder="Password"
      type="password"
      value="Password123!"
    />
    <input
      className="filled"
      id="confirmPassword"
      name="confirmPassword"
      onChange={[Function]}
      placeholder="Repeat Password"
      type="password"
      value="Password123!"
    />
    <div
      className="spacer"
    />
    <input
      className="submit-button"
      id="submit"
      onClick={[Function]}
      type="submit"
      value="Change Password"
    />
  </form>
</Fragment>
`;

exports[`Submit password change and respond with success 1`] = `
<Fragment>
  <div
    className="logo-container"
  >
    <a
      href="https://buildingim.com"
      rel="noopener noreferrer"
      target="_blank"
    >
      <img
        alt="BuildingIM logo"
        src="/images/logos/buildingim-logo.png"
      />
    </a>
  </div>
  <div
    className="welcome-text-heading"
  >
    Welcome
  </div>
  <div
    className="welcome-text-body"
  >
    Please enter a new password for your account.
  </div>
  <div
    className="auth-form"
    style={
      Object {
        "textAlign": "center",
      }
    }
  >
    <h3>
      Your password has been successfully changed.
    </h3>
    <input
      className="submit-button"
      id="submit"
      onClick={[Function]}
      type="submit"
      value="Back to login page"
    />
    /&gt;
  </div>
</Fragment>
`;

exports[`Submit password change but passwords don't match 1`] = `
<Fragment>
  <div
    className="logo-container"
  >
    <a
      href="https://buildingim.com"
      rel="noopener noreferrer"
      target="_blank"
    >
      <img
        alt="BuildingIM logo"
        src="/images/logos/buildingim-logo.png"
      />
    </a>
  </div>
  <div
    className="welcome-text-heading"
  >
    Welcome
  </div>
  <div
    className="welcome-text-body"
  >
    Please enter a new password for your account.
  </div>
  <div
    className="error-callout"
  >
    <div>
      Passwords do not match
    </div>
  </div>
  <form
    className="sign-in-form form"
    id="reset-password-form"
    onSubmit={[Function]}
  >
    <input
      className="filled"
      id="password"
      name="password"
      onChange={[Function]}
      placeholder="Password"
      type="password"
      value="These do"
    />
    <input
      className="filled"
      id="confirmPassword"
      name="confirmPassword"
      onChange={[Function]}
      placeholder="Repeat Password"
      type="password"
      value="not match"
    />
    <div
      className="spacer"
    />
    <input
      className="submit-button"
      id="submit"
      onClick={[Function]}
      type="submit"
      value="Change Password"
    />
  </form>
</Fragment>
`;
